### PR TITLE
[Listener] Process ProxyCreation events

### PR DESCRIPTION
**Pull request by franckc**
### Description:
 - Modified listener to watch for ProxyCreation event emitted by the ProxyFactory contract
 - Upon receiving an event, the listener looks up the owner (e.g. wallet) of the proxy and stores the mapping proxy address <-> wallet address in the DB
 
This proxy table can then be used for analytics purposes, but also for Origin Rewards, in order to credit to the wallet action recorded under the proxy address. For example, a listing sold or identity publish action performed via the proxy will get attributed to the wallet that owns that proxy address.

### Checklist:

- [X] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: infra/discovery/src/listener/listener.js [hunk 1]
```diff
@@ -115,8 +117,9 @@ async function main() {
   const context = await new Context().init(config, errorCounter)
 
   const contracts = {
-    IdentityEvents: contractsContext.identityEvents,
-    Marketplace: contractsContext.marketplace
+    //IdentityEvents: contractsContext.identityEvents,
```
Comment (nick) [inline infra/discovery/src/listener/listener.js]: Need to uncomment these again?

File: packages/graphql/src/contracts.js [hunk 2]
```diff
@@ -182,6 +182,8 @@ export function setNetwork(net, customConfig) {
 
   setMarketplace(config.V00_Marketplace, config.V00_Marketplace_Epoch)
   setIdentityEvents(config.IdentityEvents, config.IdentityEvents_Epoch)
+  setProxyFactory(config.ProxyFactory, config.ProxyFactory_Epoch)
+  // TBD: do we need those 2 lines ?
```
Comment (nick) [inline packages/graphql/src/contracts.js]: it's used in utils/proxy.js but we could refer the context.proxyFactory there instead now we're setting it here

Comment (franckc) [reply packages/graphql/src/contracts.js]: Cleaned up. Sorry I had no pushed my latest changes.

File: infra/discovery/src/listener/listener.js (hunk 1 above)
Comment (franckc) [reply infra/discovery/src/listener/listener.js]: Done.

**Pull request merged**